source/core/slang-linked-list.h

```diff
@@ -276,6 +270,22 @@ template <typename T> class LinkedList
             n2->prev = n1;
         else
             tail = n1;
```

Comment:
should set both n->prev and n->next to null since it is already removed from the list.